swagger.yaml

```diff
@@ -0,0 +1,249 @@
+swagger: "2.0"
+
+info:
+  title: "Tweet API"
+  description: "This provides an API for creating a message (post), reading a message (get), editing a message (put), delete a message (delete)."
+  version: "1.0"
+
+host: localhost
+
+tags:
+- name: "tweets"
+
+schemes:
+  - "https"
+
+consumes:
+  - application/json
+produces:
+  - application/json
+
+paths:
+
+  /tweets:
+  
+    get:
+      tags:
+      - tweets
+      description: "search for a user's tweets"
+      produces:
+      - application/json
+
+      parameters:
+      - in: "query"
+        name: userName
+        required: false
+        type: string
+
+      - in: "query"
+        name: tweetId
+        required: false
+        type: integer
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/tweetsData"
+        400:
+          description: The specified user ID is invalid
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: page not found
```

Comment:
All of our APIs adhere to JSON API specification. In [JSON API](http://jsonapi.org/format/#crud-creating-responses-404), `404` stands for "related resources not found" instead of a page not found. In your example, `/tweets` is the resource itself and it shouldn't be not found in any case. (If there is no "result" founded, we should return an empty array with `200` HTTP code instead).